grpc-protocol/src/main/java/com/linecorp/armeria/client/grpc/protocol/UnaryGrpcClient.java

```diff
@@ -76,12 +76,13 @@
 
     /**
      * Constructs a {@link UnaryGrpcClient} for the given {@link WebClient}.
+     *
+     * <p>Prefer using a standard client building pattern, e.g.:
+     * <pre>{@code
+     * UnaryGrpcClient client = Clients.builder("gproto+http://127.0.0.1:8080").build(UnaryGrpcClient.class);
+     * }</pre>.
+     * </p>
      */
-    // TODO(anuraaga): We would ideally use our standard client building pattern, i.e.,
-    // Clients.builder(...).build(UnaryGrpcClient.class), but that requires mapping protocol schemes to media
-    // types, which cannot be duplicated. As this and normal gproto+ clients must use the same media type, we
-    // cannot currently implement this without rethinking / refactoring core and punt for now since this is an
-    // advanced API.
     public UnaryGrpcClient(WebClient webClient) {
```

Comment:
I considered marking these constructors as deprecated, but they are still used inside the client factory. If we want this client to follow the same pattern as WebClient (where WebClient.class is passed to the `ClientFactory.build` but doesn't have a public constructor) we'll need to change `UnaryGrpcClient` to become a public interface and have a `DefaultUnaryGrpcClient` implementation - that would require a larger refactoring and might be an overkill, so for now I kept it as is. Let me know if you think it should be deprecated or you have other suggestions @trustin @ikhoon @minwoox

Reply:
I prefer to deprecate the public constructors and remove them in Armeria 2.0.
Though it is an unstable API, I think many users already use `UnaryGrpcClient`.